lib/Dancer/Plugin/SimpleCRUD.pm

```diff
@@ -610,10 +610,27 @@ CONFIRMDELETE
             my $dbh = database($args{db_connection_name});
             my $where = _get_where_filter_from_args(\%args);
             $where->{$key_column} = $id;
-            $dbh->quick_delete($table_name, $where)
+
+            my %params = params;
+            my $meta_for_hook = {
+                args => \%args,
+                params => \%params,
+                table_name => $table_name,
+                key_column => $key_column,
+            };
+            # fire the pre-delete hook, in case user wants to perform trickery before the delete
+            execute_hook('delete_row_pre_delete', $meta_for_hook );
```

Comment:
I wonder if there should be a way for the hook to block deletion of the row?  It seems to me that a pre-delete hook would mostly be useful for "double-check it's OK to delete this, and prevent it if not".  Granted, the hook could probably output a message and `halt()`?